src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,109 @@
 package co.inventorsoft.academy.collections.model;
 
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.Set;
+import java.util.*;
+import java.util.function.Function;
+
+public class Range<T extends Comperable> implements Set<T> {
+
+    private ArrayList<T> elementData  = new ArrayList<>();
+    private T to;
+    private T from;
+
+    public Range(T from, T to, Function<T, T> function) {
+        if(from.compareTo(to)>0){
+            return;
```

Comment:
don't return, just use **else**. Try to use **return** statement only on the last row of the method (if need)

Reply:
fixed!